generator-scrivito/generators/obj/index.js

```diff
@@ -0,0 +1,49 @@
+const Generator = require("yeoman-generator");
+const lodash = require("lodash");
+
+module.exports = class extends Generator {
+  start() {
+    this.prompt({
+      type: "input",
+      name: "objClassName",
+      message: "Enter the name of the Obj (e.g. MyPage):",
+    }).then(answers => {
+      const objClassName = answers.objClassName;
+      const humanFriendlyName = lodash.startCase(objClassName);
+      this._writeXObjComponent(objClassName);
+      this._writeXEditingConfig(objClassName, humanFriendlyName);
+      this._writeXObjClass(objClassName);
```

Comment:
The three methods `_writeX...` are very similar. How about extracting the common code into a helper method like this?

```js
  _generateFile(templatePath, destinationPath, context) {
    this.fs.copyTpl(
      this.templatePath(templatePath),
      this.destinationPath(destinationPath),
      context
    );
  }
```

Then in the `start` method you can first create a `folder` const like this:

```js
const folder = `src/Objs/${objClassName}`;
```

And then generate the wanted files:

```js
      this._generateFile(
        "XObjComponent.js.ejs",
        `${folder}/${objClassName}Component.js`,
        { objClassName }
      );
      this._generateFile(
        "XEditingConfig.js.ejs",
        `${folder}/${objClassName}EditingConfig.js`,
        { objClassName, humanFriendlyName }
      );
      this._generateFile(
        "XObjClass.js.ejs",
        `${folder}/${objClassName}ObjClass.js`,
        { objClassName }
      );
```

As always this is a trade-off, since the `start` method now has more lines. But for a reader it because much faster clear, what the `start` method actually does.

If you like my suggestion please also apply this in `widget/index.js`.